src/EventStore.Core/Data/VNodeState.cs

```diff
@@ -1,21 +1,22 @@
 namespace EventStore.Core.Data {
+	//WARNING: new states must be added at the bottom of the enum otherwise it may break cluster and client compatibility
 	public enum VNodeState {
-		Initializing,
-		DiscoverLeader,
-		Unknown,
-		PreReplica,
-		CatchingUp,
-		Clone,
-		Follower,
-		PreLeader,
-		Leader,
-		Manager,
-		ShuttingDown,
-		Shutdown,
-		ReadOnlyLeaderless,
-		PreReadOnlyReplica,
-		ReadOnlyReplica,
-		ResigningLeader
+		Initializing = 0,
+		DiscoverLeader = 1,
```

Comment:
Is there any particular reason we slotted `DiscoverLeader` as `1` instead of adding it to the end?

Reply:
@pgermishuys yes, the main reason I've added it at the top is because I've tried to keep it in the logical order of states that a node goes through when starting up

Reply:
it also doesn't break anything right now, but new states will need to be added to the bottom after the next major release.

Reply:
Thanks @shaan1337. The reasoning makes sense.